lua/wikis/marvelrivals/Infobox/Person/Player/Custom.lua

```diff
@@ -0,0 +1,180 @@
+	flex = {category = 'Flex Players', variable = 'Flex', isplayer = true},
+	strategist = {category = 'Strategist Players', variable = 'Strategist', isplayer = true},
+	vanguard = {category = 'Vanguard Players', variable = 'Vanguard', isplayer = true},
+
+	-- Staff and Talents
+	analyst = {category = 'Analysts', variable = 'Analyst', isplayer = false},
+	observer = {category = 'Observers', variable = 'Observer', isplayer = false},
+	host = {category = 'Hosts', variable = 'Host', isplayer = false},
+	coach = {category = 'Coaches', variable = 'Coach', isplayer = false},
+	caster = {category = 'Casters', variable = 'Caster', isplayer = false},
+	talent = {category = 'Talents', variable = 'Talent', isplayer = false},
+	producer = {category = 'Producers', variable = 'Producer', isplayer = false},
+	streamer = {category = 'Streamers', variable = 'Streamer', isplayer = false},
+}
+
+local SIZE_HERO = '25x25px'
+local MAX_NUMBER_OF_SIGNATURE_HEROES = 3
+
+---@class MarvelRivalsInfoboxPlayer: Person
+---@field role {category: string, variable: string, isplayer: boolean?}?
+---@field role2 {category: string, variable: string, isplayer: boolean?}?
+local CustomPlayer = Class.new(Player)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return Html
+function CustomPlayer.run(frame)
+	local player = CustomPlayer(frame)
+	player:setWidgetInjector(CustomInjector(player))
+
+	player.role = player:_getRoleData(player.args.role)
+	player.role2 = player:_getRoleData(player.args.role2)
+
+	return player:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local caller = self.caller
+	local args = caller.args
+
+	if id == 'custom' then
+		local heroes = Array.sub(caller:getAllArgsForBase(args, 'hero'), 1, MAX_NUMBER_OF_SIGNATURE_HEROES)
+		local heroIcons = Array.map(heroes, function(hero)
+			return CharacterIcon.Icon{character = CharacterNames[hero:lower()], size = SIZE_HERO}
+		end)
+
+		Array.appendWith(widgets,
+			Cell{
+				name = #heroIcons > 1 and 'Signature Heroes' or 'Signature Hero',
+				content = {table.concat(heroIcons, '&nbsp;')},
+			}
+		)
+	elseif id == 'role' then
+		return {
+			Cell{name = 'Role', content = {
+				caller:_displayRole(caller.role),
+				caller:_displayRole(caller.role2),
+			}},
+		}
+	end
+
+	return widgets
+end
+
+---@param role string?
+---@return {category: string, variable: string, isplayer: boolean?}?
+function CustomPlayer:_getRoleData(role)
+	return ROLES[(role or ''):lower()]
+end
+
+---@param roleData {category: string, variable: string, isplayer: boolean?}?
+---@return string?
+function CustomPlayer:_displayRole(roleData)
+	if not roleData then return end
+
+	return Link{
+		link = ':Category:' .. roleData.category,
+		children = roleData.variable
+	}
+end
+
+---@param args table
+function CustomPlayer:defineCustomPageVariables(args)
+	Variables.varDefine('role', (self.role or {}).variable)

```

Comment:
what are those wiki vars needed for?

Reply:
Honestly, i highly doubt the wiki even knows why they are setting those vars. They copied the module from Overwatch (and in fairness, OW likely doesn't need it either)